packages/api/src/controllers/stream.ts

```diff
@@ -442,32 +486,39 @@ app.get("/:parentId/sessions", authorizer({}), async (req, res) => {
     }
   }
 
-  let [sessions] = await db.stream.find(query, {
+  let [sessions] = await db.session.find(query, {
     order: `data->'lastSeen' DESC NULLS LAST`,
     limit,
     cursor,
   });
 
   const ingest = await getIngestBase(req);
-  sessions = sessions.map((session) => {
-    session = withRecordingFields(ingest, session, !!forceUrl);
-    if (!raw) {
-      if (session.previousSessions && session.previousSessions.length) {
-        session.id = session.previousSessions[0]; // return id of the first session object so
-        // user always see same id for the 'user' session
-      }
-      const combinedStats = getCombinedStats(
+  sessions = await Promise.all(
+    sessions.map(async (session) => {
+      session = await withRecordingFields(
+        req.config,
+        ingest,
         session,
-        session.previousStats || {}
+        !!forceUrl
       );
-      return {
-        ...session,
-        ...combinedStats,
-        createdAt: session.userSessionCreatedAt || session.createdAt,
-      };
-    }
-    return session;
-  });
+      if (!raw) {
+        if (session.previousSessions && session.previousSessions.length) {
+          session.id = session.previousSessions[0]; // return id of the first session object so
+          // user always see same id for the 'user' session
```

Comment:
Hmm, the only problem here is that this logic depends on some deprecated fields on the old child stream objects that were used as sessions and recordings.

I think we need some further compatibility logic here. Specifically, here we should probably identify that the `session` is from the old pipeline and treat it the same way as before, by fetching the `db.stream` child stream instead. I'm 99% sure that we have a guarantee that every `db.session` from the old pipeline had a corresponding `db.stream` object with the same ID, so here we just need to fetch it using the same ID. So the full thing looks like:
 - check if it is a new or old session type. If we don't have that yet, we should probably create a field for that, maybe even just a `version` wdyt? Would prefer to use something implicit though to avoid handling an additional version field.
 - if it's a new session, call `withRecordingFields` on it and return (this whole `raw` logic is only valid on old old pipeline, when we merged child streams by timestamp)
 - if it's an old session, fetch the child stream and go through this old logic of `raw`. Also call `withRecordingFields` but I'd even just fake that it is a session with an `as any` just to simplify things here (if the code works).

WDYT? I wrote a lot but I think the actual change is small, just trying to explain it.